### DISCUSSION TITLE
How to prevent logs from being created?

### DISCUSSION THREAD

**Discussion by 4DeepThought2** [Serene-Arc/bulk-downloader-for-reddit#353] [Q&A] [answered]
How can I prevent BDFR creating log folders/files?  I am not interested in them.

**Comment by aliparlakci** [comment, marked as answer]
Actually, no user is expected be interested in the log files. They are there for us, developers. We use them to identify the bugs users encounter. We request the log files for each bug report.

However, as oppose to the old versions (before v2) of the project, log files are kept in a separate directory, away from your downloads. Also, there is a limit for maximum number of log files there can exists at the same time. So, they won't bloat your hard drive.

Answer to your questions is solid no and it will probably stay as no forever. Because allowing such action would prevent BDFR to develop and improve.

**Reply by 4DeepThought2** [reply]
Thanks, Ill look at editing the code locally then.  Right now I just run a shell script that recursively goes through the directories and deletes the logs.  The issue isn't bloating, I don't care about space.  The issue is when the mounted directories are presented to SMB, dealing with some 10,000+ folders and even more files.  Any reduction in number of objects helps.  Access times for disk queuing significantly decreases when I remove logs.  Thanks for the reply.  Just needed to know if it was inherently possible, and you provided the answer.

**Reply by aliparlakci** [reply]
I am a bit lost in your reply. Currently, there is only one folder that contains log files and its location is explain here: https://github.com/aliparlakci/bulk-downloader-for-reddit#configuration . Also, there can only be 3 log files at max by default. 

If one directory and three files still causing that much performance issue to you, you can run a single command to delete the file `log_output.txt` after each run of bdfr.

**Reply by Serene-Arc** [reply]
Yeah, there's zero possibility that there's ever going to be any option to disable the logfiles. There are methods to restrict their size, and they have many uses. They're actually quite useful as sources of information about the run, as evidenced by the `scripts/` examples that show you how to extract stats and so on.

@4DeepThought2 You may be confused between the logfiles from v1 and v2. v1 had logfiles in the directory of the download and afaik did not have any methods to control their size. v2 puts all logfiles in the configuratiohn directory for the user in an OS-specific matter and, as aliparlakci said, they rollover automatically and are restricted to one current file and three past runs: four files max.
